src/ixdat/measurements.py

```diff
@@ -0,0 +1,375 @@
+        MS datasets, for example) and appending (sequential EC datasets). Either way,
+        all the raw series (or their placeholders) are just stored in the lists.
+        TODO: Make sure with tests this is okay, differentiate using | operator if not.
+        """
+        obj_as_dict = self.as_dict()
+        new_name = self.name + " AND " + other.name
+        new_technique = self.technique + " AND " + other.technique
+
+        from .techniques import TECHNIQUE_CLASSES
+
+        if new_technique in TECHNIQUE_CLASSES:
+            cls = TECHNIQUE_CLASSES[new_technique]
+        elif self.__class__ is other.__class__:
+            cls = self.__class__
+        else:
+            cls = Measurement
+
+        new_series_list = self.series_list
+        new_series_list.append(other.series_list)
+        new_component_measurements = self.component_measurements or [self]
+        new_component_measurements.append(other.component_measurements or [other])
+        obj_as_dict.update(
+            name=new_name,
+            series_list=new_series_list,
+            component_measurements=new_component_measurements,
+        )
+        return cls.from_dict(obj_as_dict)
+
+
+#  ------- Now come a few module-level functions for series manipulation ---------
+#  TODO: move to a .build module or similar
+
+
+def append_series(series_list):
+    """Return series appending series_list relative to series_list[0].tseries.tstamp"""
+    s0 = series_list[0]
+    if isinstance(s0, TimeSeries):
+        return append_tseries(series_list)
+    elif isinstance(s0, TimeSeries):
+        return append_tseries(series_list)
+    raise BuildError(
+        f"An algorithm of append_series for series like {s0} is not yet implemented"
+    )
+
+
+def append_vseries_by_time(series_list):
+    """Return new series with the data in series_list appended"""
+    name = series_list[0].name
+    cls = series_list[0].__class__
+    unit = series_list[0].unit
+    data = np.array([])
+
+    for s in series_list:
+        if not (s.name == name and s.unit == unit and s.__class__ == cls):
+            raise BuildError(f"can't append {series_list}")
+        data = np.append(data, s.data)
+
+    tseries = append_tseries([s.tseries for s in series_list])
+    vseries = cls(name=name, unit=unit, data=data, tseries=tseries)
+    return vseries
+
+
+def append_tseries(series_list, tstamp=None):
+    """Return new TSeries with the data appended relative to series_list[0].tstamp"""
+    name = series_list[0].name
+    cls = series_list[0].__class__
+    unit = series_list[0].unit
+    tstamp = tstamp or series_list[0].tstamp
+    data = np.array([])
+
+    for s in series_list:
+        if not (s.name == name and s.unit == unit and s.__class__ == cls):
+            raise BuildError(f"can't append {series_list}")
+        data = np.append(data, s.data + s.tstamp - tstamp)
+
+    tseries = cls(name=name, unit=unit, data=data, tstamp=tstamp)
```

Comment:
One line

Reply:
Is there any efficiency to be gained, or is it just stylistic? Sometimes I think it helps readability to be reminded that what I'm returning is indeed a vseries.
Changed to one line, in any case.